integration-tests/scripts/wait-for-release.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+#
+# Summary:
+#   Monitors a Kubernetes Release custom resource until it reaches a terminal state
+#   (Succeeded or Failed). It periodically fetches the Release status, displays
+#   a summary of its various processing stages (Tenant Collectors, Managed Collectors,
+#   Tenant Pipeline, Managed Pipeline, Final Pipeline), and provides console URLs
+#   for associated PipelineRuns. If the Release fails, it attempts to fetch and
+#   display logs from the relevant PipelineRuns.
+#
+# Parameters:
+#   None.
+#
+# Environment Variables:
+#   RELEASE_NAME      - The name of the Release custom resource to monitor. Required.
+#   RELEASE_NAMESPACE - The Kubernetes namespace where the Release resource exists. Required.
+#   CONSOLEURL        - (Derived by the script) The base URL for the OpenShift console,
+#                       used to construct links to PipelineRuns and Releases.
+#                       It's fetched from the 'pipelines-as-code' ConfigMap.
+#   APPLICATION       - (Derived by the script) The name of the application associated
+#                       with the Release, extracted from the Release's labels.
+#
+# Dependencies:
+#   kubectl, jq, tkn (for fetching PipelineRun logs), oc (for fetching CONSOLEURL).
+
+function getPipelinerunFromStatus() { # args are json, statusSection
+  json=$1
+  statusSection=$2
+
+  filter=".status."$statusSection".pipelineRun"
+  PLR=$(jq -r $filter <<< "${json}")
+  echo ${PLR//null/}
+}
+
+function getConsoleLogForRelease() { # args are application, ns, name
+  app=$1
+  ns=$2
+  name=$3
+  releaseUrl="${CONSOLEURL}ns/${ns}/applications/${app}/releases/${name}"
+  echo "${releaseUrl}"
+}
+
+function getConsoleLogFromPLR() { # args are json, statusSection
```

Comment:
The pipelineRun log is important for debugging. We may need to worry about the completeness and deplay of the log if it's got from console.

Reply:
We do not use the UI to gather logs. We only display the console URLs in the event a developer is watching a live test run or wants to examine the logs long after a run has finished. This is possible since Tekton Results keeps the logs despite the pipelinerun being deleted.

Reply:
If you are not going to collect the logs further, the PipelineRuns and applications need to be left for debugging. To give CI a relatively clean environment, can we reduce the number of runs for the test? Such as, triggering it one or two times by a Cronjob instead of triggering it by each PR.

Reply:
When the PLRs are complete, tekon results stores the PLR metadata and logs in a db. Then the PLR gets deleted.

if you know the url (which we do since we print it), we can always go look at the logs afterwards.

Reply:
What gives you the impression or idea that the script is not cleaning up?

The script starts and creates unique resources and makes note of their names. Same for branches

The script has a trap function such that any exit of the script will delete those unique resources and branches.

If, for whatever reason, a run gets aborted or killed by the cluster and the trap is not executed, then we can make use of the manual cleanup scripts.